pageserver: logging tweaks

jcsp opened the pull request:
- The `Attaching tenant` log  message omitted some useful information like the generation and mode
- info-level messages about writing configuration files were unnecessarily verbose
- During process shutdown, we don't emit logs about the various phases: this is very cheap to log since we do it once per process lifetime, and is helpful when figuring out where something got stuck during a hang.

arpad-m commented on pageserver/src/tenant/mgr.rs:
```diff
@@ -587,7 +587,13 @@ pub(crate) fn tenant_spawn(
-    info!("Attaching tenant {tenant_shard_id}");
+    info!(
+        tenant_id = %tenant_shard_id.tenant_id,
+        shard_id = %tenant_shard_id.shard_slug(),
```
Do we need separate tenant and shard id? couldn't we just have one param for `tenant_shard_id`?

koivunej replied:
We did discuss alternatives at least with Christian; we cannot yet use valuable (cfg tracing-unstable) nor would it give us this (selected code fragment) so the other alternative was to create custom macros around all tracing macros, which did not seem too great either, or use a proc macro to do similar which did not seem productive either.

jcsp replied:
The idea of having separate labels is to enable people to confidently filter on tenant_id without having to remember to also search for tenant_shard_id and/or include a shard wildcard -- that convention is new, was established in https://github.com/neondatabase/neon/pull/5957

arpad-m replied:
`tenant_shard_id` has both a display and a debug impl, I don't see the problem? This would be similar to [one of my early PRs](https://github.com/neondatabase/neon/pull/4630/files#diff-74152cb7b3681c454c4156d530ce544ee8f15048c9f3cacc736dba5dc7c0d664R4543).

koivunej replied:
> confidently filter on tenant_id

and

> tenant_shard_id has both a display and a debug impl, I don't see the problem?

I think it's the confidence, as in you remain confident to filter logs by:

`tenant_id=XYZ ` vs. `tenant_shard_id=XYZ\S+ ` I guess ... I would not have had a problem.

> This would be similar to [one of my early PRs](https://github.com/neondatabase/neon/pull/4630/files#diff-74152cb7b3681c454c4156d530ce544ee8f15048c9f3cacc736dba5dc7c0d664R4543).

I don't see the connection to there; you were adding a one way for layers to be printed in messages/parameters and using it. Now we use the layer file names everywhere thanks to that :+1:.

arpad-m replied:
tbh I never filter logs by `tenant_id=XYZ`, I always just filter by `XYZ`.

arpad-m replied:
(but I understand that some use cases might need higher precision than what I am doing, and in those instances being able to filter via `tenant_id=XYZ` makes sense)

arpad-m commented:
Regarding the "persisting tenantconf", it seems to be hit kinda rarely. I looked at a random tenant in grafana and it only had that log line once in the last 30 days.

koivunej commented on pageserver/src/lib.rs:
```diff
@@ -53,6 +53,7 @@ pub use crate::metrics::preinitialize_metrics;
 
 #[tracing::instrument(skip_all, fields(%exit_code))]
 pub async fn shutdown_pageserver(deletion_queue: Option<DeletionQueue>, exit_code: i32) {
+    tracing::info!("shutdown: starting shutdown...");
```
does it make sense to prefix these as there is already a `shutdown_pageserver` span?

koivunej replied:
Perhaps you are adding the prefix to mask some other problem?

jcsp replied:
These are gone now.

jcsp commented:
>
Regarding the "persisting tenantconf", it seems to be hit kinda rarely. I looked at a random tenant in grafana and it only had that log line once in the last 30 days.

Right - this is something that was behind a Generation::is_none check, and started getting emitted once generations were enabled (you can see the actual volume of messages in staging).

koivunej commented on pageserver/src/tenant.rs:
```diff
@@ -2503,7 +2503,7 @@ impl Tenant {
         }
 
-        info!("persisting tenantconf to {config_path}");
+        debug!("persisting tenantconf to {config_path}");
```
Ah finally, I've been so annoyed by this, but thought it changed a bit with the generations so have been keeping quiet.

Excellent that we can finally let go, logging full paths is an bad idea anyways.

koivunej replied:
Hmm ok there's a consensus that this is rare, we must have a duplicate then somewhere.

arpad-m replied:
> logging full paths is an bad idea anyways.

I disagree, it was helpful for me during debugging. Same goes for full URLs. You don't have to manually reconstruct the path but can actually directly copy it from the log and then inspect the file should the need arise.

koivunej replied:
In the context of tests, it can quickly give you the path so you don't need to combine test_output directory, but in the prod/staging logs they are just noise as we have just one path for everything.

koivunej replied:
I don't understand why do you think this is rare, staging is full of them:

```
2023-12-05T13:30:51.263652Z  INFO init_tenant_mgr:persist_tenant_config{tenant_id=3717abd2d0bd5266f9d031b24b6cf9fe shard_id=0000}:persist_tenant_config_at{tenant_id=3717abd2d0bd5266f9d031b24b6cf9fe shard_id=0000}: persisting tenantconf to /storage/pageserver/data/tenants/3717abd2d0bd5266f9d031b24b6cf9fe/config-v1
2023-12-05T13:30:51.265080Z  INFO init_tenant_mgr:persist_tenant_config{tenant_id=735fc681d2cc55acdc89a9c43882f60c shard_id=0000}:persist_tenant_config_at{tenant_id=735fc681d2cc55acdc89a9c43882f60c shard_id=0000}:persist_tenant_config_legacy{tenant_id=735fc681d2cc55acdc89a9c43882f60c shard_id=0000}: persisting tenantconf to /storage/pageserver/data/tenants/735fc681d2cc55acdc89a9c43882f60c/config
```

So is prod:

```
2023-12-05T13:17:32.381120Z  INFO request{method=POST path=/v1/tenant/ request_id=2eb7d9745f22b893cb6223686598df26}:tenant_create{tenant_id=X}:persist_tenant_config_at{tenant_id=Y}:persist_tenant_config_legacy{tenant_id=X}: persisting tenantconf to /storage/pageserver/data/tenants/c265ae5f7c0040ad226f0ebdc693b38f.___temp/config
```

koivunej commented on pageserver/src/lib.rs:
```diff
@@ -62,6 +63,7 @@ pub async fn shutdown_pageserver(deletion_queue: Option<DeletionQueue>, exit_cod
         Duration::from_secs(1),
     )
     .await;
+    tracing::info!("shutdown: page service listeners stopped.");
```
note that the timed already logs completions, so each of these would now print two messages for the same one thing.

koivunej reviewed (changes requested):
Let's not add duplicate log messages for shutdown steps completing.

jcsp commented:
So what threw me about the existing log messages from timed() is that they had a `task=` parameter.  So they looked like they were logging the shutdown of a task, rather than a logical step.  I've removed the commit that added the separate logs for each of those steps, and added a commit that calls the parameter `stage=` instead of task, to avoid confusing it with tokio tasks.
